duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -313,6 +321,27 @@ private String canonQueryString()
     return Util.join(args.toArray(), "&");
   }
 
+  private String canonXDuoHeaders(){
+    ArrayList<String> canonList = new ArrayList<String>();
+    for (String name : additionalHeaders.keySet()){
+      String value = additionalHeaders.get(name);
+      canonList.add(name + value);
+    }
+    return Util.join(canonList.toArray(), String.valueOf(Character.MIN_VALUE));
```

Comment:
Do we need to sort the headers by name?

Reply:
That's a good point! I totally forget about it. I will add it in the `canon_header` method

Reply:
I used [TreeMap](https://docs.oracle.com/javase/8/docs/api/java/util/TreeMap.html) for `additionalDuoHeaders`, from [oracle doc](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html) the default Comparator is lexicographic-order.

Reply:
perfecto